### PR TITLE
 Amplitude DAG: Save JSONL in ingest_amplitude_raw_* bucket

### LINKED ISSUE #1240: Reorganize Amplitude data in GCS
Part of the work for #960.

See [this comment](https://github.com/cal-itp/data-infra/pull/1239#issuecomment-1076666810) from @atvaccaro for details

## Tasks

* [x] Update the [operator](https://github.com/cal-itp/data-infra/blob/feat-amplitude-operator/airflow/plugins/operators/amplitude_to_flattened_json.py#L123) to pass a `bucket` parameter, equal to `ingest_amplitude_raw_dev` or `ingest_amplitude_raw_prod` depending on the value of `is_development()`
* [x] Update the [loader](https://github.com/cal-itp/data-infra/blob/feat-amplitude-operator/airflow/dags/amplitude_benefits_loader/jsonl_bigquery.sql#L69) to load from the new bucket

### PR DESCRIPTION
**Pull request by machikoyasuda**
# Overall Description

close #1240 part of #960 #1116 

- Save `jsonl` file from Amplitude into new `ingest_amplitude_raw_dev` or `ingest_amplitude_raw_prod` bucket
image

- Read `jsonl` file from new buckets
- Add `is_development` function to macros, so that operators can use it

## Checklist for all PRs

- [x] Run `pre-commit run --all-files` to make sure markdown/lint passes
- [x] Link this pull request to all issues that it will close using keywords (see GitHub docs about [Linking a pull request to an issue using a keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)). Also mention any issues that are partially addressed or are related.

## Airflow DAG changes checklist

- [x] Include this section whenever any change to a DAG in the `airflow/dags` folder occurs, otherwise please omit this section.
- [x] Verify that all affected DAG tasks were able to run in a local environment
- [x] Take a screenshot of the graph view of the affected DAG in the local environment showing that all affected DAG tasks completed successfully
- [x] Add/update documentation in the `docs/airflow` folder as needed
- [x] Fill out the following section describing what DAG tasks were added/updated

image
image

This PR updates the `amplitude_benefits` DAG in order to....
- Update `api_to_jsonl`: Save the JSONL file to `gs://ingest_amplitude_raw_dev` or `gs://ingest_amplitude_raw_prod` bucket
- Update `jsonl_bigquery `: Get the JSONL file from `gs://ingest_amplitude_raw_dev` or `gs://ingest_amplitude_raw_prod` bucket, as determined by `is_development()` macro, as defined in [airflow/dags/macros.py](https://github.com/cal-itp/data-infra/pull/1279/files#diff-a9dc672975f537be6f0ec7ec039c377c09ee4fa15c24bed81c7306ccd0753308)

### CODE REVIEW

**Review by thekaveman — CHANGES_REQUESTED** [review]

File: airflow/plugins/operators/amplitude_to_flattened_json.py [hunk 1]
```diff
@@ -119,5 +120,11 @@ def execute(self, context):
         )
         gcs_file_path = f"amplitude/{self.app_name}/{start}-{end}.jsonl"
```
[the remark this answers is not in the document]
Comment (machikoyasuda) [reply airflow/plugins/operators/amplitude_to_flattened_json.py]: Updated:

Bucket path - 
image


URI path -
image

**Review by thekaveman — APPROVED** [review]
🚀

**Review by angela-tran — APPROVED** [review]
👍

**Comment by thekaveman** [comment]
I linked the issue after seeing #1188 remain open even after the PR was merged. But I think it's because the commits have to go into the default branch (`main`) before Issues are auto-closed.

**Pull request merged**